src/hotspot/share/oops/klass.hpp

```diff
@@ -514,15 +514,16 @@ class Klass : public Metadata {
   // Want a pattern to quickly diff against layout header in register
   // find something less clever!
   static int layout_helper_boolean_diffbit() {
-    jint zlh = array_layout_helper(T_BOOLEAN);
-    jint blh = array_layout_helper(T_BYTE);
-    assert(zlh != blh, "array layout helpers must differ");
-    int diffbit = 1;
-    while ((diffbit & (zlh ^ blh)) == 0 && (diffbit & zlh) == 0) {
-      diffbit <<= 1;
-      assert(diffbit != 0, "make sure T_BOOLEAN has a different bit than T_BYTE");
-    }
-    return diffbit;
+    uint zlh = checked_cast<uint>(array_layout_helper(T_BOOLEAN));
+    uint blh = checked_cast<uint>(array_layout_helper(T_BYTE));
```

Comment:
Use of check_cast is probably wrong. I think an alh is negative. Oops, my mistake. It probably doesn't fail currently because of [JDK-8314258](https://bugs.openjdk.org/browse/JDK-8314258).

Reply:
Note that by "my mistake" I meant it was a mistake to use `checked_cast` here.